pkg/ipcache/kvstore.go

```diff
@@ -312,7 +312,8 @@ func (iw *IPIdentityWatcher) OnUpdate(k storepkg.Key) {
 	}
 
 	peerIdentity := ipIDPair.ID
-	if option.Config.EnableRemoteNodeIdentity && peerIdentity == identity.ReservedIdentityHost {
+	if peerIdentity == identity.ReservedIdentityHost {
+		//if option.Config.EnableRemoteNodeIdentity && peerIdentity == identity.ReservedIdentityHost {
```

Comment:
```suggestion
```

Reply:
Though I wonder if we should just remove the if statement. After all, we don't expect remote nodes to announce themselves as "host" anymore.

Reply:
True!

Reply:
>Though I wonder if we should just remove the if statement. After all, we don't expect remote nodes to announce themselves as "host" anymore.

@gandro Where is this logic? I can't find it. I can only find that [`UpsertIPToKVStore` appears to upsert host IPs with unmodified IDs](https://github.com/cilium/cilium/blob/687e4f8f403aa75f3e1e42005db51b29fdd9e5ef/pkg/endpoint/policy.go#L836-L850). Am I missing something?

Reply:
@nathanjsweet Yeah what I said was wrong. I looked at the [old migration docs](https://docs.cilium.io/en/v1.7/install/upgrade/#configmap-remote-node-identity) and they mentioned (the long removed) `transmit-host-src`, which I thought meant impacted IPCache entries, but it doesn't. It only affected the identities transmitted in VXLAN headers.

However, I'm not sure what host IPs actually announce in the linked endpoint sync controller. 

I assume external workloads will announce regular identity that is different for `reserved:host` (i.e. the one read by `e.SecurityIdentity.ID` allocated when the EW joins the cluster). And for the host endpoint I believe we don't run the `runIPIdentitySync` controller either?

In any case, I'm happy to keep the if statement there, it's a good safeguard against bugs in the remote agents in case they ever attempt to create an kvstore entry with host identity.